text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,324 @@
+
+Since the [TypeScript] compiler does not conform to [SemVer], future releases of
+the `jsii` compiler will not conform to [SemVer] either. In line with the
+[TypeScript] compiler versioning scheme, breaking changes may be introduced in
+any new release that updates the `major` or `minor` version, but **not** when
+only the `patch` level changed.
+
+This change enables `jsii` users to benefit from the latest and greatest
+features introduced in the [TypeScript] language as well as from performance
+improvements and bug fixes introduced in recent versions of [TypeScript],
+without requiring the entire ecosystem to update compiler versions at the same
+time.
+
+### Why should I use this feature?
+
+The new `jsii` versioning strategy gives developers more control over the
+[TypeScript] language version they are developing, without forcing an update
+schedule on them.
+
+Developers are free to decide when it is appropriate for them to upgrade their
+`jsii` dependency to a new `major.minor` version line, bringing in new
+[TypeScript] language features as well as performance improvements and bug
+fixes.
+
+The `jsii` compiler makes the due dilligence to ensure the artifacts produced
+(in particular, the `.d.ts` declarations files and the `.jsii` assemblies)
+remain compatible with previous versions, so that consumers of libraries need
+not update at the same time as their dependencies.
+
+### What happens to `jsii` v1?
+
+Version `1.0.0` of the `jsii` compiler was released over two years ago (in
+February 2020), and will transition to the *Maintenance* tier of the
+[AWS SDKs and Tools maintenance policy][aws-policy] 6 months after this release.
+
+During this initial 6 months period, `jsii` will continue to receive full
+support, including bug fixes and feature support, as it has in the past two
+years.
+
+Once it enters the *Maintenance* tier, it will only receive fixes for critical
+bugs and security issues. The `v1` release line will remain in *Maintenace* for
+12 months, after which it will transition into the *End-of-Life* tier.
+
+### What is the support policy for these new releases?
+
+Starting with release `4.7.0`, new features will only be added to the *current*
+release line (the *latest* `major.minor` stream), which corresponds to the
+currently active [TypeScript] release line.
+
+Critical bug fixes and security patches will be provided for previous release
+lines for 12 months, which covers 3 or 4 previous lines. Features will not be
+back-ported to non-current release lines.
+
+Previous release lines will be considered *end-of-life* 12 months after they
+were superceded by a new *current* line.
+
+## Internal FAQ
+
+### How will we maintain backwards compatibility?
+
+The change only affects the [TypeScript] compiler version used internally by
+`jsii` and the versioning scheme for the `jsii` package itself.
+
+The compiler will continue to emit `.jsii` assemblies that conform to the schema
```

Comment:
Are we committed to not breaking that?

Reply:
For the foreseeable future, I don't think there is much of a way around that.